### PR TITLE
Registry based treatment phases

### PR DESCRIPTION
**Pull request by alintreznai**
This PR implements registry based patient stages.

### CODE REVIEW

File: rdrf/registry/patients/admin_forms.py [hunk 1]
```diff
@@ -154,7 +154,19 @@ def save(self, commit=True):
...
         model = PatientStage
-        fields = "__all__"
+        fields = ["registry", "name", "allowed_prev_stages", "allowed_next_stages"]
+
+    def __init__(self, *args, **kwargs):
+        super(PatientStageForm, self).__init__(*args, **kwargs)
+        if self.instance.pk:
+            self.fields['registry'].disabled = True
+        registry = self.instance.registry or None
+        stage_qs = (
```
Comment (sztamas) [inline rdrf/registry/patients/admin_forms.py]: `registry__isnull=True` is the same as `registry=None`, so you can just use `registry=registry` ie. you don't need an `if` at all.

File: rdrf/registry/patients/models.py [hunk 2]
```diff
@@ -1126,6 +1126,7 @@ def family(self):
 
 
 class PatientStage(models.Model):
+    registry = models.ForeignKey(Registry, on_delete=models.SET_NULL, null=True)
```
Comment (sztamas) [inline rdrf/registry/patients/models.py]: shouldn't this be on_delete cascade?

File: rdrf/registry/patients/admin_forms.py (hunk 1 above)
Comment (alintreznai) [reply rdrf/registry/patients/admin_forms.py]: Updated.

File: rdrf/registry/patients/models.py (hunk 2 above)
Comment (alintreznai) [reply rdrf/registry/patients/models.py]: Updated.

File: rdrf/rdrf/templates/admin/patients/patientstage/change_form.html [hunk 3]
```diff
@@ -0,0 +1,97 @@
+{% extends "rdrf_cdes/base.html" %}
```
Comment (sztamas) [inline rdrf/rdrf/templates/admin/patients/patientstage/change_form.html]: Can we extend from `admin/change_form.html` instead and add just the bit we need so we don't copy the whole file?

File: rdrf/rdrf/templates/admin/patients/patientstage/change_form.html [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+{% endif %}
+
+{# JavaScript for prepopulated fields #}
+{% prepopulated_fields_js %}
+
+<script language="JavaScript">
+    $('#id_registry').change(function() {
+        var registry_id = this.value ? this.value: -1;
+        $.ajax({
+            url: "/" +registry_id + "/patientstages/",
```
Comment (sztamas) [inline rdrf/rdrf/templates/admin/patients/patientstage/change_form.html]: 1. the url should come from a var set by Django templates so you can look up the URL by name instead
2. I don't think I like the looks of that URL when you don't have a registry specified. Can we just use `patientstages` **followed** by an _optional_ registry_id?

File: rdrf/rdrf/templates/admin/patients/patientstage/change_form.html (hunk 3 above)
Comment (alintreznai) [reply rdrf/rdrf/templates/admin/patients/patientstage/change_form.html]: Updated.

File: rdrf/rdrf/templates/admin/patients/patientstage/change_form.html (hunk 4 above)
Comment (alintreznai) [reply rdrf/rdrf/templates/admin/patients/patientstage/change_form.html]: Updated.

**Comment by alintreznai** [comment]
Closing this as it's not needed for now. Keeping the branch for future reference.

**Pull request closed without merging**